activerecord/lib/active_record/insert_all.rb

```diff
@@ -177,5 +178,10 @@ def quote_columns(columns)
             columns.map(&connection.method(:quote_column_name))
           end
       end
+
+      PrimaryKeyIndexDescriptor = Struct.new(:columns) do # :nodoc:
+        def where
+        end
+      end
```

Comment:
🤔 I guess I was thinking that I didn't have/need all the information to instantiate an `ActiveRecord::ConnectionAdapters::IndexDefinition` when a duck-type would work just as well, but looking at it again I see the initializer sets sane defaults for all the missing information, so perhaps it would be better to just use it as you suggest? I'll give it a try. 👍